File: core/execution_reality/semantic_execution_pressure_engine.py
from __future__ import annotations

from typing import Any, Dict


MAX_PRESSURE = 100000


def compute_execution_pressure(
    runtime_ir: Dict[str, Any],
) -> Dict[str, Any]:

    transitions = list(
        runtime_ir.get(
            "transitions",
            [],
        )
    )

    event_stream = runtime_ir.get(
        "event_stream",
        {},
    )

    events = list(
        event_stream.get(
            "events",
            [],
        )
        if isinstance(event_stream, dict)
        else []
    )

    pressure_score = min(
        len(transitions)
        + len(events),
        MAX_PRESSURE,
    )

    return {
        "pressure_score": pressure_score,
        "bounded": True,
    }

File: src/execution_reality/semanticExecutionPressureEngine.ts
/**
 * Converted from Python: core/execution_reality/semantic_execution_pressure_engine.py
 * @generated — WebWeaveX python→javascript library port
 */

import * as py from "../runtime/pyCompat.js";

export let MAX_PRESSURE: any = 100000;
export function computeExecutionPressure(runtime_ir: any): any {
  var transitions: any = [...py.iter(py.get(runtime_ir, "transitions", []))];
  var event_stream: any = py.get(runtime_ir, "event_stream", {});
  var events: any = [...py.iter((((event_stream !== null && typeof event_stream === "object" && !Array.isArray(event_stream) && !(event_stream instanceof Set) && !(event_stream instanceof Map))) ? py.get(event_stream, "events", []) : []))];
  var pressure_score: any = py.min([py.add(py.len(transitions), py.len(events)), MAX_PRESSURE]);
  return {"pressure_score": pressure_score, "bounded": true};
}
